shared/containers/profile.container.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { retrieveProfile } from 'shared/reducers/profile/profile.actions';
+import { updateUI } from 'shared/reducers/ui/ui.actions';
+
+
+const mapStateToProps = state => ({
+  auth: state.auth,
+  profile: state.profile,
+  location: state.location,
+  ui: state.ui,
+});
+
+const mapDispatchToProps = dispatch => ({
+  retrieveProfile: (user_id) => {
+    retrieveProfile.assignTo(dispatch);
+    retrieveProfile(user_id);
+  },
+  updateUI: (payload) => {
+    updateUI.assignTo(dispatch);
+    updateUI(payload);
+  },
+});
```

Comment:
I originally made a mistake using assignTo. All you really need here is `dispatch(updateUI(payload))`. 

There can be some funky side effects to using assignTo, but don't worry about changing this until we actually discover a specific issue related to this.